### ISSUE TITLE
Error when starting the virtual camera

### ISSUE THREAD

**Issue by ringods** [obsproject/obs-studio#11026]
### Operating System Info

macOS 14

### OBS Studio Version

30.2.2

### OBS Studio Log URL

https://obsproject.com/logs/YbHJAwvX3hRB66IK

### Expected Behavior

Pressing the `Start Virtual Camera` button should enable this function

### Current Behavior

When I press the `Start Virtual Camera` button, I get a dialog that there was an error installing the virtual camera.

<img width="372" alt="Screenshot 2024-07-24 at 16 04 19" src="https://github.com/user-attachments/assets/c88fcf80-f92a-46fd-81ff-4a24d9d404c7">

### Steps to Reproduce

1. Launch OBS Studio v30.2.2
2. Press `Start Virtual Camera`
3. error dialog appears

### Anything else we should know?

OBS Studio is installed via the Homebrew Cask:

https://github.com/Homebrew/homebrew-cask/blob/master/Casks/o/obs.rb

**Comment by infinitewarp** [comment]
I encountered a similar error trying to enable the virtual cam on macOS 14 (Sonoma) after updating to 30.2.2 today using OBS's built-in updater.

![Screenshot 2024-07-24 at 11 24 15 AM](https://github.com/user-attachments/assets/197c652c-f9c0-40d3-8349-0c490e1f8081)

I tried flipping permissions off and on in System Settings, restarting OBS multiple times, and reviewing OBS's permissions, but none of that helped. The only thing that _did_ resolve the issue for me was manually dragging OBS to the trash, confirming the dialog that warned it would remove a system extension, downloading and installing a fresh copy of OBS from obsproject.com, starting it up, and approving the system extension permissions again.

TLDR: Delete and reinstall OBS. 

I guess something didn't actually update correctly from the auto-updater, but it's not clear to me. 🤷

**Comment by RytoEX** [comment]
@infinitewarp 
Your issue (no virtual camera after updating) is almost certainly the same as #10381. As far as I know, it has nothing to do with the updater.

@ringods 
If this happened after an OBS Studio update, then this is a duplicate of #10381. If this happened on a fresh install of OBS Studio, then that may be #9912 or something different. Please let us know specifically how you installed OBS Studio 30.2.2.

**Comment by ringods** [comment]
@RytoEX this indeed happened after an OBS Studio update. However, when I updated (via Homebrew) to v30.2.0, I got the error from #10381. Reinstalling from a download from the main website solved this for me. Today, I had another update to v30.2.2 and got the error I reported. Given the other dialog, I have the impression it is something different. I leave that up to you as authors/developers of the product though.

**Comment by jcm93** [comment]
@ringods for clarity, is the issue you currently have after updating to 30.2.2 the error code 8 error, or the "not found" error? And exactly what method did you use to update (manual check for updates in-app, automatic in-app, homebrew, etc.)

A system extension error code 8 indicates an invalid code signature or missing entitlements, so this is definitely distinct from the "not found" issue. I suspect installing from Homebrew has something to do with this but not able to say for sure until I have a chance to try to reproduce.

**Comment by ringods** [comment]
I updated via Homebrew on macOS and got the error code 8 message.

```
systemextensionsctl list
6 extension(s)
--- com.apple.system_extension.cmio
enabled	active	teamID	bundleID (version)	name	[state]
*	*	2MMRE5MTB8	com.obsproject.obs-studio.mac-camera-extension (30.2.1/10049553606)	com.obsproject.obs-studio.mac-camera-extension	  [activated enabled]
		2MMRE5MTB8	com.obsproject.obs-studio.mac-camera-extension (30.1.2/8576208847)	com.obsproject.obs-studio.mac-camera-extension	  [terminated waiting to uninstall on reboot]
		2MMRE5MTB8	com.obsproject.obs-studio.mac-camera-extension (30.2.0/9914953239)	com.obsproject.obs-studio.mac-camera-extension	  [terminated waiting to uninstall on reboot]
```

**Comment by jcm93** [comment]
> systemextensionsctl list
> 6 extension(s)
> --- com.apple.system_extension.cmio
> enabled	active	teamID	bundleID (version)	name	[state]
> *	*	2MMRE5MTB8	com.obsproject.obs-studio.mac-camera-extension (30.2.1/10049553606)	com.obsproject.obs-studio.mac-camera-extension	  [activated enabled]
> 		2MMRE5MTB8	com.obsproject.obs-studio.mac-camera-extension (30.1.2/8576208847)	com.obsproject.obs-studio.mac-camera-extension	  [terminated waiting to uninstall on reboot]
> 		2MMRE5MTB8	com.obsproject.obs-studio.mac-camera-extension (30.2.0/9914953239)	com.obsproject.obs-studio.mac-camera-extension	  [terminated waiting to uninstall on reboot]

Thanks. Could you also start up OBS and post the log from that startup?

**Comment by jcm93** [comment]
Hmm. Can you check if this issue persists after restarting your machine?

**Comment by ringods** [comment]
@jcm93 That would be the same/similar as the log link posted in my initial message. I have this problem on both my M1 and Intel Mac.

**Comment by jcm93** [comment]
> @jcm93 That would be the same/similar as the log link posted in my initial message. I have this problem on both my M1 and Intel Mac.

Yes, apologies, I realized that and deleted my question before you replied.

> I just rebooted my Intel Mac but the problem persists.

Now that you've rebooted, can you post the output of `systemextensionsctl list` one more time as well as a new log from OBS?

**Comment by ringods** [comment]
@jcm93 since the extensions list in my previous comment came from my M1 Mac, I also rebooted that machine. Here is the refreshed output:

```
--- com.apple.system_extension.cmio
enabled	active	teamID	bundleID (version)	name	[state]
*	*	2MMRE5MTB8	com.obsproject.obs-studio.mac-camera-extension (30.2.1/10049553606)	com.obsproject.obs-studio.mac-camera-extension	[activated enabled]
```

Log: https://obsproject.com/logs/uTyxmMk5jv1Fhm9v

**Comment by MrNobodyX3** [comment]
I get this issue where it wants me to accept a privacy issue but it doesn't exist in settings.

<img width="1036" alt="image" src="https://github.com/user-attachments/assets/d9b078c3-8022-4db3-b006-d0cf1f4d04e1">

**Comment by HellShuro** [comment]
I encountered the same problem, which makes it impossible for us to use. When can this problem be fixed?

**Comment by zzzzz167** [comment]
> 我遇到这个问题，它希望我接受隐私问题，但它在设置中不存在。
> 
> <img alt="图像" width="1036" src="https://private-user-images.githubusercontent.com/39644589/356786326-d9b078c3-8022-4db3-b006-d0cf1f4d04e1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ3MjY5NTMsIm5iZiI6MTcyNDcyNjY1MywicGF0aCI6Ii8zOTY0NDU4OS8zNTY3ODYzMjYtZDliMDc4YzMtODAyMi00ZGIzLWIwMDYtZDBjZjFmNGQwNGUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI3VDAyNDQxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFhZmJiMmY2YWQzMmMxZmQzODk4ZjRlMDI3MWZiZmY5ZjJjNTU2NDg1M2RhOWZmMzUyMDRiZjNiMjBhMWMyMDImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.z-iJ9DkWRtRyB-JMHLkS1PLH-CLBZ7FvCNWvG9FjGTU">

I have the same problem so what should i do to us virtual camera again?

**Comment by zzzzz167** [comment]
> I get this issue where it wants me to accept a privacy issue but it doesn't exist in settings.
> 
> <img alt="image" width="1036" src="https://private-user-images.githubusercontent.com/39644589/356786326-d9b078c3-8022-4db3-b006-d0cf1f4d04e1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ3Njk0NjQsIm5iZiI6MTcyNDc2OTE2NCwicGF0aCI6Ii8zOTY0NDU4OS8zNTY3ODYzMjYtZDliMDc4YzMtODAyMi00ZGIzLWIwMDYtZDBjZjFmNGQwNGUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI3VDE0MzI0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM1ZDFkYjI0ZDA5Y2JlNjY4YzIwMDYxMjJlMGU3YzlhZmFkYjZmOGQzZmJkZDVhYWEwNGU3YWRlMmRkOGQwNDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.63IPhp6_WLro3DDyz0QBDbNG30XHrm02yZ-QZx0lHKA">

@MrNobodyX3 
I have the same problem as you
When I use the `systemextensionsctl list`. It outputs the following information:
![image](https://github.com/user-attachments/assets/1524211e-3eda-4d9b-b357-1270038b3f59)
So it seems that this extension option has moved in some newer versions of the system and then I found the right spot and turned on the virtual camera at the System Settings > General > Login Items & Extensions > Camera Extensions.
I'm not sure if this will work for the problems you're describing, but it's worth a try.

**Comment by MrNobodyX3** [comment]
> >  So it seems that this extension option has moved in some newer versions of the system and then I found the right spot and turned on the virtual camera at the System Settings > General > Login Items & Extensions > Camera Extensions. I'm not sure if this will work for the problems you're describing, but it's worth a try.

This did resolve my issue. Thank you so much.

**Comment by ringods** [comment]
@zzzzz167 I can't navigate to this path in the System Settings:

`System Settings > General > Login Items & Extensions > Camera Extensions`

Under `General`, I only have `Login Items`. But there is no trace of `Camera Extensions`. Am I missing something?

**Comment by jcm93** [comment]
`System Settings > General > Login Items & Extensions > Camera Extension` is the location under macOS 15.

`System Settings > Privacy & Security > scroll down to Security` is the location pre-macOS 15.

FWIW, I've tried to reproduce this error installing OBS with homebrew and subsequently updating, but could not. A system extension error code of 8 still indicates a real issue though, that we should try get to the bottom of.

I have heard one report via another channel that removing the `~/Library/Application Support/obs-studio` directory resolved this issue. This did not make sense to me, but may nevertheless be worth a shot on your end if you are still seeing this specific error.

**Comment by el-hoshino** [comment]
I have a similar problem, except in my case it's error 10 🤔 

![スクリーンショット 2024-11-08 14 53 47](https://github.com/user-attachments/assets/f78e18af-857a-41b3-823e-ce4ce6b5bf84)

**Comment by jtprogru** [comment]
I have a similar problem

<img width="263" alt="Image" src="https://github.com/user-attachments/assets/23cd8ffc-710e-4593-a65d-eb05b8aa43f7" />

**Comment by jcm93** [comment]
This issue doesn't seem to be able to be reproduced outside of Homebrew installations; the best guess at the moment is that it's an internal macOS issue. For affected users, the extension bundle signature is actually valid according to the system, and when queried, the system actually reports a correct and valid version of the extension being installed. macOS returns the signing validation error when loading the extension for unclear reasons.

Homebrew also isn't actually an official release channel for OBS and the formula isn't maintained by the project itself. This issue might receive more attention if anyone here runs into it with OBS installed via an official, supported release channel.

**Comment by jtprogru** [comment]
This problem solved after uninstall OBS from Homebrew and installing from DMG image downloaded from official site.

**Comment by agzam** [comment]
~~Nope, nothing is fixing this issue for me.~~ 

- I tried removing OBS, 
- I tried re-installing it from scratch (downloading from the website), 
- I tried removing ~/Library/Application Support/obs-studio. 
- I tried rebooting the computer. 
- I checked all the permissions
- I tried `sudo cp -R /Applications/OBS.app/Contents/Resources/obs-mac-virtualcam.plugin /Library/CoreMediaIO/Plug-Ins/DAL/` - because that folder is empty, and I thought I need to copy it there.
- I tried `sudo codesign --remove-signature /Library/CoreMediaIO/Plug-Ins/DAL/obs-mac-virtualcam.plugin/Contents/MacOS/obs-mac-virtualcam` 

None of that has helped and I can't figure it out. OBS 31.1.2 (64 bit). Sequoia on M1 Mac. 


----
Oh, okay. I figured it out. I did this:

`sudo spctl --master-disable`

and it told me to:

"Globally disabling the assessment system needs to be confirmed in System Settings."

I'm not even sure I had to run that command, I think I could've just gone here:

<img height="520" alt="Image" src="https://github.com/user-attachments/assets/136eafda-95d1-4317-b785-6d57c0a2e851" />

Then "Start Virtual Camera" installs the plugin that needs to be enabled in `General -> Login Items  & Extensions -> Camera Extensions`. And I had to restart OBS and it worked.

**Comment by maxvisser** [comment]
I also got the issue OBS 32.0.2 (64 bit). Sequoia on M1 Mac., what @agzam proposes did not work for me. I installed 30.2.3 from github, and then it finally installed the OBS virtualcam plugin
